doc/source/text.rst

```diff
@@ -201,9 +207,106 @@ and optional groups like
 
 .. ipython:: python
 
-   pd.Series(['a1', 'b2', '3']).str.extract('(?P<letter>[ab])?(?P<digit>\d)')
+   pd.Series(['a1', 'b2', '3']).str.extract('([ab])?(\d)')
```

Comment:
make ``extract`` and ``extractall`` sub-sections (I think you might have to use ``^^^^``) as the sub-headings